File: frontend/src/services/quizController.js
import api from './api';

const QuizController = {
  async getRecommendations() {
    const res = await api.get('/api/quiz/recommendations');
    return res.data;
  },

  async getHistory() {
    const res = await api.get('/api/quiz/history');
    return res.data;
  },

  async createQuiz({ subject = 'code tutor', difficulty = 'beginner', quizType = 'mixed', totalQuestions = 5, timeLimit = 600 } = {}) {
    const res = await api.post('/api/quiz/create', {
      subject,
      difficulty,
      quiz_type: quizType,
      total_questions: totalQuestions,
      time_limit: timeLimit,
    });
    return res.data;
  },

  async getQuestions(quizId) {
    const res = await api.get(`/api/quiz/${quizId}/questions`);
    return res.data;
  },

  async submitQuiz({ quizId, answers = [], totalTimeTaken = 0 }) {
    const res = await api.post('/api/quiz/submit', {
      quiz_id: quizId,
      answers,
      total_time_taken: totalTimeTaken,
    });
    return res.data;
  }
};

export default QuizController;

File: mobile-app/lib/src/controllers/quiz_controller.dart
import 'package:code_mentor_mobile/src/controllers/chat_controller.dart';

class QuizController {
  QuizController._();

  static final QuizController instance = QuizController._();

  final _chat = ChatController.instance;

  Future<Map<String, dynamic>> getRecommendations() {
    return _chat.getQuizRecommendations();
  }

  Future<Map<String, dynamic>> getHistory() {
    return _chat.getQuizHistory();
  }

  Future<Map<String, dynamic>> createQuiz({
    required String subject,
    String difficulty = 'beginner',
    String quizType = 'mixed',
    int totalQuestions = 5,
    int timeLimit = 600,
  }) {
    return _chat.createQuiz(
      subject: subject,
      difficulty: difficulty,
      quizType: quizType,
      totalQuestions: totalQuestions,
      timeLimit: timeLimit,
    );
  }

  Future<Map<String, dynamic>> getQuestions(int quizId) {
    return _chat.getQuizQuestions(quizId);
  }

  Future<Map<String, dynamic>> submitQuiz({
    required int quizId,
    required List<Map<String, dynamic>> answers,
    int totalTimeTaken = 0,
  }) {
    return _chat.submitQuiz(
      quizId: quizId,
      answers: answers,
      totalTimeTaken: totalTimeTaken,
    );
  }
}
